### PR TITLE
Add Google analytics

### PR DESCRIPTION
**Pull request by tsingh777**
https://github.com/KhalisFoundation/sttm-desktop/issues/131

Initial tracking for setting changes and searches. 

Uses https://www.npmjs.com/package/universal-analytics module 

Followed this guide https://kilianvalkhof.com/2018/apps/using-google-analytics-to-gather-usage-statistics-in-electron/

Example of output:
image

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const ua = require('universal-analytics'); // https://www.npmjs.com/package/universal-analytics
+const isOnline = require('is-online');
+
+const trackingId = 'UA-129669333-1';
```
Comment (tsingh777) [inline analytics.js]: This is one I set up, need to put in a khalis one or look it up somehow.

**Comment by navdeepsinghkhalsa** [comment]
@tsingh777 There is an `in progress` tag; would you like for me to review yet?

**Comment by tsingh777** [comment]
@navdeepsinghkhalsa yes please. we shouldn't merge this until the we agree on the tracking ID to be used.

**Comment by navdeepsinghkhalsa** [comment]
I'm happy with the implementation strategy, though we need an opt-out :)
@maneetpaul Jee, could you please work with @tsingh777 on which events you would like to track?

**Comment by tsingh777** [comment]
does this violate GDPR? how should we handle the opt-out @saintsoldierx  @maneetpaul

**Comment by navdeepsinghkhalsa** [comment]
It doesn't violate GDPR, as everything is anonymous (we don't even collect machine-related info 👍) but just a checkbox so a user can opt-out should suffice. This is our implementation in Sundar Gutka. 
image 2018-12-05 09 08 43

**Comment by ManjotS** [comment]
lets add  a  setting  for analytics (see  sundar  gutka settings)

**Comment by tsingh777** [comment]
added Opt out
image

**Comment by navdeepsinghkhalsa** [comment]
It looks good to me. Could you please update the UA ID?

**Pull request merged**